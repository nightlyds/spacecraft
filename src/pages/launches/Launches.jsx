import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import MenuButton from "../../basic/components/menu/MenuButton.jsx";
import Menu from "../../basic/components/menu/Menu.jsx";
import counterpart from "counterpart";
import en from "./lang/en";
import uk from "./lang/uk";
import "./css/launches.css";
import "./css/media_launches.css";
import Loading from "../../basic/components/Loading.jsx";
import Error from "../../basic/components/Error.jsx";
import Footer from "../../basic/components/Footer.jsx";
import LanguageChange from "../../basic/components/LanguageChange.jsx";
import LaunchesList from "./components/launchesList/LaunchesList.jsx";
import LaunchesDefault from "./components/launchesList/LaunchesDefault.jsx";
import OneLaunch from "./components/oneLaunch/OneLaunch.jsx";

counterpart.registerTranslations("en", en);
counterpart.registerTranslations("uk", uk);

const Launches = ({
  loadUpLaunches,
  loadUpOneLaunch,
  oneLaunch,
  launches,
  loading,
  error,
}) => {
  let [menuOpenCheck, setMenuOpen] = useState(false);
  let [oneLaunchOpenCheck, setOneLaunchOpen] = useState(false);

  useEffect(() => {
    loadUpLaunches();
  }, []);

  function menuOpen() {
    setMenuOpen(!menuOpenCheck);
  }

  function oneLaunchOpen(launch) {
    setOneLaunchOpen(!oneLaunchOpenCheck);
    if (launch) {
      loadUpOneLaunch(launch);
    }
  }
  return (
    <div className="launches">
      {oneLaunchOpenCheck && (
        <OneLaunch launch={oneLaunch} oneLaunchClose={oneLaunchOpen} />
      )}
      <MenuButton menuOpen={menuOpen} icon={menuOpenCheck} />
      {menuOpenCheck && <Menu />}
      <LanguageChange />
      <div className="launches-content">
        <div className="launches-list-box">
          {loading && !error && <LaunchesDefault />}
          {loading && !error && (
            <LaunchesList launches={launches} oneLaunchOpen={oneLaunchOpen} />
          )}
        </div>
      </div>
      {!loading && <Loading />}
      {loading && error && <Error />}
      <Footer />
    </div>
  );
};

Launches.propTypes = {
  loadUpLaunches: PropTypes.func,
  loadUpOneLaunch: PropTypes.func,
  oneLaunch: PropTypes.object,
  launches: PropTypes.object,
  loading: PropTypes.bool,
  error: PropTypes.bool,
};

Launches.defaultProps = {
  loadUpLaunches: () => {},
  loadUpOneLaunch: () => {},
  oneLaunch: {},
  launches: {},
  loading: false,
  error: false,
};

export default Launches;
