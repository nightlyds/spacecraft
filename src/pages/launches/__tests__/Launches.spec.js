import React from "react";
import Launches from "../Launches.jsx";
import { shallow } from "enzyme";

const setUp = (props) => shallow(<Launches {...props} />);

describe("Launches component tests", () => {
  describe("Snapshot", () => {
    let component;

    beforeEach(() => {
      component = setUp();
    });

    it("Launches default snapshot", () => {
      expect(component).toMatchSnapshot();
    });

    it("Launches data snapshot", () => {
      component = setUp({ launches: { 0: "0", 1: "1" }, loading: true });
      expect(component).toMatchSnapshot();
    });

    it("Launches error snapshot", () => {
      component = setUp({ loading: true, error: true });
      expect(component).toMatchSnapshot();
    });
  });

  describe("Props", () => {
    let component;

    it("loadUpLaunches props test", () => {
      component = Launches.defaultProps.loadUpLaunches();
      expect(component).toBe(undefined);
    });

    it("loadUpOneLaunch props test", () => {
      component = Launches.defaultProps.loadUpOneLaunch();
      expect(component).toBe(undefined);
    });

    it("oneLaunch props test", () => {
      component = Launches.defaultProps.oneLaunch;
      expect(component).toMatchObject({});
    });

    it("launches props test", () => {
      component = Launches.defaultProps.launches;
      expect(component).toMatchObject({});
    });

    it("loading props test", () => {
      component = Launches.defaultProps.loading;
      expect(component).toBe(false);
    });

    it("error props test", () => {
      component = Launches.defaultProps.error;
      expect(component).toBe(false);
    });
  });
});
