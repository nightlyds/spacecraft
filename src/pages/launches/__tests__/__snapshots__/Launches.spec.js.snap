// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launches component tests Snapshot Launches data snapshot 1`] = `
<div
  className="launches"
>
  <MenuButton
    icon={false}
    menuOpen={[Function]}
  />
  <LanguageChange />
  <div
    className="launches-content"
  >
    <div
      className="launches-list-box"
    >
      <LaunchesDefault />
      <LaunchesList
        launches={
          Object {
            "0": "0",
            "1": "1",
          }
        }
        oneLaunchOpen={[Function]}
      />
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Launches component tests Snapshot Launches default snapshot 1`] = `
<div
  className="launches"
>
  <MenuButton
    icon={false}
    menuOpen={[Function]}
  />
  <LanguageChange />
  <div
    className="launches-content"
  >
    <div
      className="launches-list-box"
    />
  </div>
  <Loading />
  <Footer />
</div>
`;

exports[`Launches component tests Snapshot Launches error snapshot 1`] = `
<div
  className="launches"
>
  <MenuButton
    icon={false}
    menuOpen={[Function]}
  />
  <LanguageChange />
  <div
    className="launches-content"
  >
    <div
      className="launches-list-box"
    />
  </div>
  <Error />
  <Footer />
</div>
`;
